doc/yast-invoking-external-commands.md

```diff
@@ -0,0 +1,205 @@
+# Invoking External Commands in YaST
+
+## From the Ruby Code
+
+```Ruby
+ret_code = SCR.Execute(path(".target.bash"), command)
+```
+
+or
+
+```Ruby
+result = SCR.Execute(path(".target.bash_output"), command)
+ret_code = result["exit"]
+stdout = result["stdout"]
+stderr = result["stderr"]
+```
+
+or
+
+```Ruby
+output = SCR.Execute(path(".target.bash_output"), command)["stdout"]
+```
+
+_(also available: .target.bash_background, .target.bash_input)_
```

Comment:
We're missing the current best practice, [Yast::Execute](https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/execute.rb), based on [Cheetah](https://github.com/openSUSE/cheetah/), a standalone gem developed at SUSE by David Majda.

Its command-spawning part is [here](https://github.com/openSUSE/cheetah/blob/ac990d5afcff45153d2d9428059cabe01feb506f/lib/cheetah.rb#L557-L559)
It uses [Kernel#exec](https://ruby-doc.com/core/Kernel.html#method-i-exec)

... which probably uses PATH but I can't find it there :-/

Reply:
AFAICS you can pass your custom environment as an option, otherwise it will keep the existing one. But in any case, afterwards it will restore the old one.

https://github.com/openSUSE/cheetah/blob/ac990d5afcff45153d2d9428059cabe01feb506f/lib/cheetah.rb#L557-L559

```Ruby
        with_env(options[:env]) do
          exec([command, command], *args)
        end
```

https://github.com/openSUSE/cheetah/blob/ac990d5afcff45153d2d9428059cabe01feb506f/lib/cheetah.rb#L437

```Ruby
    def with_env(env, &block)
      old_env = ENV.to_hash
      ENV.update(env)
      block.call
    ensure
      ENV.replace(old_env)
    end
```

Reply:
https://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run

> :env (Hash) — default: {} — Allows to update ENV for the time of running the command. if key maps to nil value it is deleted from ENV.